synapse_auto_accept_invite/__init__.py

```diff
@@ -40,10 +60,16 @@ async def on_new_event(self, event: EventBase, *args: Any) -> None:
             and event.membership == "invite"
             and self._api.is_mine(event.state_key)
         ):
-            # Make the user join the room.
-            await self._api.update_room_membership(
-                sender=event.state_key,
-                target=event.state_key,
-                room_id=event.room_id,
-                new_membership="join",
-            )
+            is_direct_message = event.content.get("is_direct", False)
+
+            # always for direct messages and if enabled for all messages (not only direct messages)
+            if (
+                is_direct_message and isinstance(is_direct_message, bool)
+            ) or not self._config.accept_invites_only_for_direct_messages:
```

Comment:
I find this condition a bit confusing. I'd reword it into:

```suggestion
            if not self._config.accept_invites_only_for_direct_messages or is_direct_message is True:
```